tools/runners/run-test-suite.py

```diff
@@ -0,0 +1,214 @@
+    parser.add_argument('--engine', metavar='FILE',
+                        help='JerryScript binary to run tests with')
+    parser.add_argument('--test-list', metavar='FILE',
+                        help='File contains test paths to run')
+    parser.add_argument('--skip-list', metavar='LIST',
+                        help='Add a comma separated list of patterns of the excluded JS-tests')
+    parser.add_argument('--test-dir', metavar='DIR',
+                        help='Directory contains tests to run')
+    parser.add_argument('--snapshot', action='store_true',
+                        help='Snapshot test')
+
+    script_args = parser.parse_args()
+    if script_args.skip_list:
+        script_args.skip_list = script_args.skip_list.split(',')
+    else:
+        script_args.skip_list = []
+
+    return script_args
+
+
+def get_tests(test_dir, test_list, skip_list):
+    tests = []
+    if test_dir:
+        tests = []
+        for root, _, files in os.walk(test_dir):
+            tests.extend([os.path.join(root, test_file) for test_file in files if test_file.endswith('.js')])
+
+    if test_list:
+        with open(test_list, "r") as test_list_fd:
+            lines = test_list_fd.read().splitlines()
+        dirname = os.path.dirname(test_list)
+        for test in lines:
+            tests.append(os.path.normpath(os.path.join(dirname, test)))
+
+    tests.sort()
+
+    def filter_tests(test):
+        for skipped in skip_list:
+            if skipped in test:
+                return False
+        return True
+
+    return [test for test in tests if filter_tests(test)]
+
+
+def get_platform_cmd_prefix():
+    if sys.platform == 'win32':
+        return ['cmd', '/S', '/C']
+    return []
+
+
+def execute_test_command(test_cmd):
+    process = subprocess.Popen(test_cmd, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, universal_newlines=True)
+    stdout = process.communicate()[0]
+    return (process.returncode, stdout)
+
+
+def main(args):
+    tests = get_tests(args.test_dir, args.test_list, args.skip_list)
+    total = len(tests)
+    if total == 0:
+        print("No test to execute.")
+        return 1
+
+    if sys.platform == 'win32':
+        original_timezone = util.get_timezone()
+        util.set_sighdl_to_reset_timezone(original_timezone)
+        util.set_timezone('UTC')
+
+    if args.snapshot:
+        passed = run_snapshot_tests(args, tests)
+    else:
+        passed = run_normal_tests(args, tests)
+
+    if sys.platform == 'win32':
+        util.set_timezone(original_timezone)
+
+    failed = total - passed
+
+    summary_list = [os.path.relpath(args.engine)]
+    if args.snapshot:
+        summary_list.append('--snapshot')
+    if args.test_dir:
+        summary_list.append(os.path.relpath(args.test_dir))
+    if args.test_list:
+        summary_list.append(os.path.relpath(args.test_list))
+    util.print_test_summary(' '.join(summary_list), total, passed, failed)
+
+    if failed > 0:
+        return 1
+    return 0
```

Comment:
Can't we just return with the `failed` value?

Reply:
We can, but ... exit code on Linux is between 0-255. If (failed%256) == 0, the exit could would be 0, which is not good. Of course, we can simplify it with bool case, but we can't return failed directly.